### DISCUSSION TITLE
Is there a way to specify included files/paths for formatting?

### DISCUSSION THREAD

**Discussion by agreedSkiing** [openrewrite/rewrite#2729] [Q&A] [answered]
Just this morning I tried to us the maven plugin to format our `src/generated/java` code but noticed that it was completely skipped when running the maven plugin version of rewrite.

For our project we check in the `src/generated/java` folder and all of the code inside of it since it generated from an openAPI specification and want to keep the formatting the same all over our internal gitlab repo.

So is it possible to set the paths for folder and files to include?

**Comment by timtebeek** [comment, marked as answer]
Hi @agreedSkiing we currently skip generated code when applying recipes in at least two places in the Maven plugin:

- https://github.com/openrewrite/rewrite-maven-plugin/blob/31701dc4a834694c118a5b5dc864aee20ae00dce/src/main/java/org/openrewrite/maven/AbstractRewriteMojo.java#L260
- https://github.com/openrewrite/rewrite-maven-plugin/blob/31701dc4a834694c118a5b5dc864aee20ae00dce/src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java#L225

And so far, the maven plugin goals only allow for adding [additional exclusions](https://github.com/openrewrite/rewrite-maven-plugin/blob/v4.39.0/src/main/java/org/openrewrite/maven/ConfigurableRewriteMojo.java#L65), not explicit inclusions.

If you feel we should support generated sources you're welcome to open an issue on [openrewrite/rewrite-maven-plugin](https://github.com/openrewrite/rewrite-maven-plugin/issues), so we can track that there.
